packages/gatsby/cache-dir/app.js

```diff
@@ -101,7 +101,7 @@ apiRunnerAsync(`onClientEntry`).then(() => {
   const renderer = apiRunner(
     `replaceHydrateFunction`,
     undefined,
-    ReactDOM.render
+    ReactDOM.hydrate
```

Comment:
This is great because it improves warnings significantly. Just to confirm, min supported version of ReactDOM for hydrate is v16 so we're good.